tests/integration/test_init.py

```diff
@@ -166,3 +167,34 @@ def test_should_overlap_3_paths():
         _overlap_path("http://testserver/first/second/third/fourth/fifth/?limit=1&offset=1", 4)
         == "http://testserver/fifth/?limit=1&offset=1"
     )
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "pagination_size,limit,expected_status",
+    [
+        (10, 10, status.HTTP_200_OK),
+        (10, 1, status.HTTP_200_OK),
+        (10, 11, status.HTTP_400_BAD_REQUEST),
+        # defaults to default limit (aka the page size on the settings)
+        (10, -1, status.HTTP_200_OK),
+        (20, 11, status.HTTP_200_OK),
+        (20, 21, status.HTTP_400_BAD_REQUEST),
+        (None, 200, status.HTTP_200_OK)
+    ],
+)
+def test_should_set_max_limit_if_received_the_header(client, pagination_size, limit, expected_status):
+    headers = {"HTTP_X_DRF_MAX_PAGINATION_SIZE": pagination_size}
+
+    response = client.get(f"/data/?limit={limit}", **headers)
+
+    assert response.status_code == expected_status
```

Comment:
**question:** Any way to guarantee that the header limit is really respected?

Reply:
I added some evidence of [tests I've done locally](https://juntossomosmais.monday.com/boards/346124957/pulses/1303133319/posts/1026121499) with the logs of the database access. I'm not sure I understand what you mean by "really respected".

Reply:
Great! But, this package is open-sourced and whoever wants to use it, won't have access to that evidence.

However, we can also leave an evidence with an automated test that verifies the response content, asserting that it doesn't return more than the specified limit. And, that's what I meant with "really respected".